src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -337,7 +337,7 @@ RCLConsensus::Adaptor::onClose(
             // pseudo-transactions
             auto validations = app_.validators().negativeUNLFilter(
                 app_.getValidations().getTrustedForLedger(
-                    prevLedger->info().parentHash));
+                    prevLedger->info().parentHash, prevLedger->seq() - 1));
```

Comment:
A unit test should be made for this. Namely, what happens if this is fed a validation with a bogus sequence?